pandas/io/parsers/base_parser.py

```diff
@@ -599,14 +604,8 @@ def _convert_to_ndarrays(
                 # type specified in dtype param or cast_type is an EA
                 if cast_type and (not is_dtype_equal(cvals, cast_type) or is_ea):
                     if not is_ea and na_count > 0:
-                        try:
-                            if is_bool_dtype(cast_type):
-                                raise ValueError(
-                                    f"Bool column has NA values in column {c}"
-                                )
-                        except (AttributeError, TypeError):
-                            # invalid input to is_bool_dtype
-                            pass
+                        if is_bool_dtype(cast_type):
+                            raise ValueError(f"Bool column has NA values in column {c}")
```

Comment:
So are there new user-facing paths where this is raised now?

Reply:
not AFAICT.  the `except (AttributeError, TypeError):` this removes should be un-reachable